src/Modules/Trinity.DynamicCluster/Trinity.Azure.Storage.Test/DownloadTest.cs

```diff
@@ -37,13 +50,21 @@ public void Init()
             dir.GetBlockBlobReference(c1.Id.ToString()).UploadFromByteArray(f1, 0, f1.Length);
             dir.GetBlockBlobReference(c2.Id.ToString()).UploadFromByteArray(f2, 0, f2.Length);
             dir.GetBlockBlobReference(Constants.c_finished).UploadText("");
+
+            var v = await m_storage.GetLatestVersion();
+
+            Assert.AreEqual(m_version, v);
         }
 
         [TestMethod]
-        public async Task GetLatestVersion()
+        public async Task Download()
         {
-            var v = await m_storage.GetLatestVersion();
-            Assert.AreEqual(m_version, v);
+            GetLatestVersion().Wait();
+            var v = await m_storage.Download(m_version, 0, 110);
+            var src = await v.DownloadAsync();
+            var src_seq = src.GetEnumerator();
```

Comment:
ps: GetEnumerator() can be implicitly invoked if you do `foreach(var buf in src_seq)`

Reply:
and thus you could say `Assert.IsTrue(Enumerable.SequentialEqual(src_seq.Single().First(), TestItem`

Reply:
Need help!  
This one passed:
```C#
var src_seq = src.GetEnumerator();
var buffer = src_seq.Current.Buffer;
Assert.IsTrue(buffer.Take(data.Length).SequenceEqual(data.Take(data.Length)));
```
However, this one cannnot work well.
```C#
var buffer = src.First().Buffer;
Assert.IsTrue(buffer.Take(data.Length).SequenceEqual(data.Take(data.Length)));
```
The second one occurs ``System.AccessViolationException at [PersistedCellEnumerator.cs, line 32](https://github.com/Microsoft/GraphEngine/blob/availability/src/Modules/Trinity.DynamicCluster/Trinity.DynamicCluster/Persistency/PersistedCellEnumerator.cs)